core/vm/instructions.go

```diff
@@ -529,10 +529,11 @@ func opCreate(instr instruction, pc *uint64, env Environment, contract *Contract
 		dataGas.Mul(dataGas, params.CreateDataGas)
 		if contract.UseGas(dataGas) {
 			env.Db().SetCode(addr, ret)
+			stack.push(addr.Big())
+		} else if params.IsHomestead(env.BlockNumber()) {
+			env.Db().Delete(addr)
+			stack.push(new(big.Int))
 		}
```

Comment:
Correct. A left-over from a previous version which had the "everything else" case outside the if clause. Another edge case we need a `StateTests` for.